src/Service/BattleEngine/BattlePhase.php

```diff
@@ -160,14 +157,14 @@ private function calculateCasualties(GameUnit $gameUnit, int $power): int
         return intval($power / $health);
     }
 
-    private function getBattlePhaseBattleStats(GameUnit $gameUnit): AbstractBattleStats
+    private function getBattlePhaseBattleStats(WorldRegionUnit $gameUnit): AbstractBattleStats
```

Comment:
Great catch! This is indeed a bug! The fix is however not sufficient, as it's not just WorldRegionUnit getting passed to this function... It's also FleetUnit.

A better fix would be to add getGameUnit() in getAttackPower and getDefensePower functions, to get the actual GameUnit from WorldRegionUnit and FleetUnit.

Reply:
Changed the fix to use the mentioned methods.